draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -5307,14 +5307,22 @@ Expect: 100-continue
    bookmarks/favorites), the user agent &MUST; either exclude the Referer header field
    or send it with a value of "about:blank".
 </t>
+<t>
+   The Referer header field value need not convey the full URI of the referring
+   page; it &MAY; be truncated to the referring origin.
+</t>
 <t>
    The Referer header field has the potential to reveal information about the request
    context or browsing history of the user, which is a privacy concern if the
    referring resource's identifier reveals personal information (such as an
    account name) or a resource that is supposed to be confidential (such as
    behind a firewall or internal to a secured service). Most general-purpose
    user agents do not send the Referer header field when the referring
-   resource is a local "file" or "data" URI. A user agent &MUST-NOT; send a
+   resource is a local "file" or "data" URI. A user agent &SHOULD-NOT; send a
+   <x:ref>Referer</x:ref> header field if the referring page was received with
+   a secure protocol, and the request target has an origin differing from that
+   of the referring page, unless the referring page explicitly allows it.
+   A user agent &MUST-NOT; send a
    <x:ref>Referer</x:ref> header field in an unsecured HTTP request if the
    referring page was received with a secure protocol.
```

Comment:
```suggestion
   referring resource was received with a secure protocol.
```

Reply:
I used `page` because that was the already existing language. If we want to change it, it should probably be changed to `representation`; you can't receive a resource (at least not in the HTTPWG; WHATWG may feel differently).

Reply:
As I said, we're talking about an indirection at some level here, but it is still the resource.

In one case, the resource asks by using Referrer-Policy, which is - I think - pretty directly the resource acting here.  In some of the other cases (`<meta>`, link decorations), the requesting is indirect in that the resource provides a representation that contains the request, but the effect is still the same.

"Page" is just the wrong abstraction.  As "resource" is used a few paragraphs up in the introduction of the field, consistency can't be the reason to use "page".

Reply:
The resource acts by sending `Referrer-Policy`, but the resource is _not_ "received". "Accessed", perhaps.

Reply:
I can live with that.  I'm just objecting to the use of "page".